src/QMCWaveFunctions/tests/test_DiracDeterminantBatched.cpp

```diff
@@ -480,4 +483,237 @@ TEST_CASE("DiracDeterminantBatched_delayed_update", "[wavefunction][fermion]")
   test_DiracDeterminantBatched_delayed_update<
       MatrixUpdateOMPTarget<ValueType, QMCTraits::QTFull::ValueType>>(2, DetMatInvertor::HOST);
 }
+
+
+#ifdef QMC_COMPLEX
+template<class DET_ENGINE>
+void test_DiracDeterminantBatched_spinor_update(const int delay_rank, DetMatInvertor matrix_inverter_kind)
+{
+
+  using ParticlePos       = ParticleSet::ParticlePos;
+  using ParticleGradient  = ParticleSet::ParticleGradient;
+  using ParticleLaplacian = ParticleSet::ParticleLaplacian;
+
+  // O2 test example from pwscf non-collinear calculation.
+  ParticleSet::ParticleLayout lattice;
+  lattice.R(0, 0) = 5.10509515;
+  lattice.R(0, 1) = -3.23993545;
+  lattice.R(0, 2) = 0.00000000;
+  lattice.R(1, 0) = 5.10509515;
+  lattice.R(1, 1) = 3.23993545;
+  lattice.R(1, 2) = 0.00000000;
+  lattice.R(2, 0) = -6.49690625;
+  lattice.R(2, 1) = 0.00000000;
+  lattice.R(2, 2) = 7.08268015;
+
+  //Shamelessly stealing this from test_einset.cpp.  3 particles though.
+  const SimulationCell simulation_cell(lattice);
+  ParticleSet ions_(simulation_cell);
+  ParticleSet elec_(simulation_cell);
+  ions_.setName("ion");
+  ions_.create({2});
+
+  ions_.R[0][0] = 0.00000000;
+  ions_.R[0][1] = 0.00000000;
+  ions_.R[0][2] = 1.08659253;
+  ions_.R[1][0] = 0.00000000;
+  ions_.R[1][1] = 0.00000000;
+  ions_.R[1][2] = -1.08659253;
```

Comment:
also here `ions_.R[0] = {0.00000000, 0.00000000, 1.08659253};` I think the xyz can be folded.

Reply:
these work fine

Reply:
should be fixed now. left off a }...need to get my vim setup to do syntax checking properly to avoid easy things like that